### PR TITLE
AP_OSD:Add avg resting cell and refactor

### PR DESCRIPTION
**Pull request by Hwurzburg**
Co-authored-by: MichelleRos

this implements the functionality of #22660 but save almost 500 bytes instead of costing flash

### CODE REVIEW

File: libraries/AP_OSD/AP_OSD.h [hunk 1]
```diff
@@ -547,6 +551,13 @@ class AP_OSD
         UNITS_AVIATION=3,
         UNITS_LAST=4,
     };
+    
+    enum {
```
Comment (tridge) [inline libraries/AP_OSD/AP_OSD.h:555]: make this an enum class

File: libraries/AP_OSD/AP_OSD_Screen.cpp [hunk 2]
```diff
@@ -1283,51 +1299,73 @@ void AP_OSD_Screen::draw_altitude(uint8_t x, uint8_t y)
     backend->write(x, y, false, "%4d%c", (int)u_scale(ALTITUDE, alt), u_icon(ALTITUDE));
 }
 
-void AP_OSD_Screen::draw_bat_volt(uint8_t x, uint8_t y)
+void AP_OSD_Screen::draw_bat_volt(uint8_t instance,uint8_t type,uint8_t x, uint8_t y)
```
Comment (tridge) [inline libraries/AP_OSD/AP_OSD_Screen.cpp:1302]: uint8_t type should be the enum class type

**Review by tridge — CHANGES_REQUESTED** [review]

**Review by tridge — APPROVED** [review]

File: libraries/AP_OSD/AP_OSD.h [hunk 3]
```diff
@@ -547,7 +558,7 @@ class AP_OSD
         UNITS_AVIATION=3,
         UNITS_LAST=4,
     };
-
+
```
Comment (peterbarker) [inline libraries/AP_OSD/AP_OSD.h:561]: whitespace aded

**Comment by tridge** [comment]
@MichelleRos can you test this? Should have your feature and saves flash

File: libraries/AP_OSD/AP_OSD_Screen.cpp [hunk 4]
```diff
@@ -1283,51 +1299,73 @@ void AP_OSD_Screen::draw_altitude(uint8_t x, uint8_t y)
...
+    float v = battery.voltage(instance);
+    float blinkvolt = osd->warn_batvolt;
     uint8_t pct;
-    if (!battery.capacity_remaining_pct(pct)) {
+    bool show_remaining_pct = battery.capacity_remaining_pct(pct);
+    uint8_t p = (100 - pct) / 16.6;
+    switch (type) {
+    case VoltageType::VOLTAGE: {
+        break;
+        }
```
Comment (peterbarker) [inline libraries/AP_OSD/AP_OSD_Screen.cpp:1313]:
```suggestion
    case VoltageType::VOLTAGE:
        break;
```

**Review by MichelleRos — APPROVED** [review]
Tested on my quad. Works well. Thanks. 🙂

File: libraries/AP_OSD/AP_OSD_Screen.cpp (hunk 4 above)
Comment (Hwurzburg) [reply libraries/AP_OSD/AP_OSD_Screen.cpp]: not according to astyle

**Pull request merged**